src/Dependencies.jl

```diff
@@ -137,16 +194,18 @@ end
 # dictionaries.  This function converts the dictionary back to the appropriate
 # AbstractDependency.
 function dependencify(d::Dict)
-    if d["type"] == "dependency"
+    if d["type"] in ("dependency", "builddependency", "hostdependency")
         uuid = isnothing(d["uuid"]) ? d["uuid"] : UUID(d["uuid"])
         version = VersionNumber(d["version-major"], d["version-minor"], d["version-patch"])
         version = version == v"0" ? nothing : version
-        return Dependency(PackageSpec(; name = d["name"], uuid = uuid, version = version))
-    elseif d["type"] == "builddependency"
-        uuid = isnothing(d["uuid"]) ? d["uuid"] : UUID(d["uuid"])
-        version = VersionNumber(d["version-major"], d["version-minor"], d["version-patch"])
-        version = version == v"0" ? nothing : version
-        return BuildDependency(PackageSpec(; name = d["name"], uuid = uuid, version = version))
+        T = if d["type"] == "dependency"
+            Dependency
+        elseif d["type"] == "builddependency"
+            BuildDependency
+        elseif d["type"] == "hostdependency"
+            HostBuildDependency
+        end
+        return T(PackageSpec(; name = d["name"], uuid = uuid, version = version))
```

Comment:
I had a moment of deja vu, where I was like "wait, I remember seeing this code before", then I realized that I implemented it the same way in my branch.  😂